roles/ycsb-bench/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+---
+# These flags are appended to the ycsb load and run commands
+- name: Set YCSB Workload Flags for Couchbase
+  set_fact:
+    ycsb_flags: "-p couchbase.host={{ couchbase_hostname }} -p couchbase.bucket=default -p couchbase.password={{ default_bucket_password }}"
+  when: ycsb.infra == "couchbase"
+
+- name: Load Data Into Database
+  k8s:
+    definition:
+      kind: Pod
+      apiVersion: v1
+      metadata:
+        name: '{{ meta.name }}-ycsb-data-load'
+        namespace: '{{ meta.namespace }}'
+        labels:
+          name: 'ycsb-load'
+      spec:
+        containers:
+        - name: bench-server
+          image: "quay.io/dustinblack/ycsb-server:testing"
+          imagePullPolicy: Always
+          command: ["/bin/sh"]
+          args: ["-c", "/ycsb/bin/ycsb load {{ ycsb.driver }} -s -P /ycsb/workloads/{{ ycsb.workload }} {{ ycsb_flags }}"]
+          #args: ["-c", "watch cat /etc/hosts"]
+  when: ycsb.workers > 0
+
+- name: Wait for Load to be Complete....
+  k8s_facts:
+    kind: Pod
+    api_version: v1
+    namespace: '{{ meta.namespace }}'
+    name: '{{ meta.name }}-ycsb-data-load'
+  register: ycsb_load_pod 
+  until: "'Completed' in (ycsb_load_pod | json_query('resources[].status.containerStatuses[].lastState.terminated.reason'))"
+  retries: 30
+  delay: 10
+
+#- name: Clean up Data Load Pod
+#  k8s:
+#    state: absent
+#    definition:
+#      kind: Pod
+#      apiVersion: v1
+#      metadata:
+#        name: '{{ meta.name }}-ycsb-data-load'
+#        namespace: '{{ meta.namespace }}'
+
+- name: Run YCSB Workload
+  k8s:
+    definition:
+      kind: Pod
+      apiVersion: v1
+      metadata:
+        name: '{{ meta.name }}-ycsb-bench'
+        namespace: '{{ meta.namespace }}'
+        labels:
+          name: 'ycsb-run'
+      spec:
+        containers:
+        - name: bench-server
+          image: "quay.io/dustinblack/ycsb-server:testing"
```

Comment:
@aakarshg @jtaleric What is the right thing to do here? This image should be built and hosted alongside the operator image, shouldn't it?

Reply:
@dustinblack I can give you access to the benchmark-operator namespace to push images to (i thought I already did)...